Decisions/scs-xxxx-v1-ssd-flavors.md

```diff
@@ -0,0 +1,249 @@
+latencies and scheduling latencies have not been observed to be an issue in
+clusters within one cloud region, the storage latency is. With remote networked
+storage as delivered from ceph, the long tail of write latency causes etcd
+to often time out heartbeats, causing a new leader election with a leader
+change, preventing control plane changes on k8s for a few seconds. Too many
+leader changes can slow down cluster operation and even bring it to a halt.
+
+The etcd requirements [are well documented](https://etcd.io/docs/v3.5/op-guide/hardware/#example-hardware-configurations).
+In particular, over a hundred of *sequential* IOPS are recommended. This
+requires write latencies in the range of a single-digit ms (or better).
+
+# Design Considerations
+
+## Options considered
+
+### One-node etcd (backed by redundant storage)
+
+If k8s uses only one control plane node, there will only be only one etcd node,
+avoiding timed out heartbeats. Single node control planes are typically not
+recommended for production workloads though. They are limited with respect
+to control plane performance, have a higher chance to fail (as a single node failure
+can create cluster control-plane downtime) and can not undergo rolling upgrades.
+
+Though not the normal setup with kubeadm, it is possible to use a multi-node
+control plane using a single-node etcd. This shares some of the challenges of
+single-node control-planes, although recovery may be faster to perform at least
+in scenarios where the etcd backend storage is redundant and not affected by the
+single-node outage.
+
+Neither scenario fulfills typical requirements for production workloads.
+
+### RAM (tmpfs) etcd storage
+
+etcd could keep its database in volatile memory (e.g. on a tmpfs filesystem).
+For multi-node etcd clusters, this would actually work, as long as at least one
+cluster member stays alive. A loss of power affecting all nodes or a hardware
+maintenance operation not tracking etcd needs would result in a complete
+loss of all cluster state. The control plane nodes would require live migration
+to avoid this in the maintenance case. For the power loss scenario, a frequent
+backup might mitigate the cluster state loss case.
+
+The etcd database is normally limited to 2GiB in size, which is something
+that is realistic to keep in main memory. (Typical database sizes are
+much smaller.)
+
+This option requires additional care and may not be suitable for all
+production scenarios, but would seem a possible fallback position.
+
+### Heartbeat slowdown
+
+To avoid causing too many fail-overs by occasional high latencies, the
+frequency of heartbeats can be lowered from the default 1/100ms.
+The reelection timeout should change along with it (typically set to
+10 beats).
+
+This will cause etcd to take a bit more time to notice the loss of a node,
+which is not typically critical if done within reasonable limits.
+This change however is not a silver bullet -- occasional write latencies
```

Comment:
This change gives no guarantee -- occasional write latencies

(reads imho better than the silver bullet wording)

Reply:
Adjusted